scipy/stats/_morestats.py

```diff
@@ -856,15 +856,21 @@ def ppcc_plot(x, a, b, dist='tukeylambda', plot=None, N=80):
 
 def _log_mean(logx):
     # compute log of mean of x from log(x)
-    return special.logsumexp(logx, axis=0) - np.log(len(logx))
+    res = special.logsumexp(logx, axis=0) - math.log(logx.shape[0])
+    return res
 
 
-def _log_var(logx):
+def _log_var(logx, xp):
     # compute log of variance of x from log(x)
     logmean = _log_mean(logx)
-    pij = np.full_like(logx, np.pi * 1j, dtype=np.complex128)
-    logxmu = special.logsumexp([logx, logmean + pij], axis=0)
-    return np.real(special.logsumexp(2 * logxmu, axis=0)) - np.log(len(logx))
+    # get complex dtype with component dtypes same as `logx` dtype;
+    # see data-apis/array-api#841
+    dtype = xp.result_type(logx.dtype, xp.complex64)
+    pij = xp.full(logx.shape, pi * 1j, dtype=dtype)
```

Comment:
This is unfortunate.

The other option is to use real `dtype` in `logsumexp` with weights (conceptually) like `b = [1, -1]`, but that comes with its own hurdles.